build/dsc/manager.rb

```diff
@@ -188,6 +188,8 @@ def document_types(markdown_file_path, dsc_types)
           end
           file.write("\n")
         end
+        
+        file.write("\nGit reference of DSC used to create types: #{ENV['DSC_REF']}")        
```

Comment:
If we're going to use an env var to store that global state, then I think we could set it after the repo is cloned if it hasn't already been set?  (I know this won't work always based on cached files, but should work some of the time).